maze_transformer/evaluation/path_evals.py

```diff
@@ -0,0 +1,109 @@
+from typing import Callable, Iterable, TypeAlias
+
+import numpy as np
+from muutils.tensor_utils import NDArray
+
+from maze_transformer.generation.constants import Coord, CoordTup
+from maze_transformer.generation.latticemaze import LatticeMaze
+
+# pylint: disable=unused-argument
+MazePath: TypeAlias = NDArray["node x_y_pos", int]
+PathEvalFunction = Callable[[LatticeMaze, MazePath, MazePath], float]
+
+
+def path_as_segments_iter(path: MazePath) -> Iterable[tuple]:
+    """
+    Iterate over the segments of a path.
+    """
+    i: int
+    n_s: Coord | CoordTup
```

Comment:
could we rename n_s and n_e to be more meaningful? is this node south and node east 🤔